README.md

```diff
@@ -796,6 +796,122 @@ The `profunctor` method takes two arguments:
 
 3. `promap` must return a value of the same Profunctor
 
+## Types
+
+In Fantasy Land, types are specified by an object containing a catamorphic
+method `cata`. `cata` has an arity matching the number of constructor functions
+belonging to each type. Each argument to `cata` is either:
+
+1. a value
```

Comment:
This should read "a value which matches the return type of `cata` itself." or something to that effect.

Reply:
Yeah that would nullify everything I said ( which is good! :D )

Reply:
Ah! That sentence confused me. But, what about removing it altogether? JS is a dynamic language and if people want to do `aMaybe.cata("foo", (n) => 12)` why should we stop them?

Reply:
> I would say the user should convert the `Maybe` to an `Either`

That only seems like a more roundabout way of doing the same thing and now you just have a "problematic" `cata` invocation on an Either that in one case returns a string and in another a number?

> I'd like to keep static type checkers (TS, Flow, etc.) in mind when specing these types out.

Agreed, but both options can be type checked in both TS and Flow.

My point is just that from the point of view of the `cata` implementation it doesn't matter what the functions return (`cata` is not allowed to inspect it). There doesn't seem to be any benefit to having that requirement in the spec.

Reply:
The spec is covered in types - if people want to do `Just(10).cata("foo", id)`, they're welcome to do so, but they're not fantasyland-compliant...